ShortenLink/Scenes/ShortenLink/ShortenLinkViewModel.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  ShortenLinkViewModel.swift
+//  ShortenLink
+//
+//  Created by Bliss on 22/11/21.
+//
+
+import Foundation
+import RxCocoa
+import RxSwift
+
+protocol ShortenLinkViewModelType {
+
+    var input: ShortenLinkViewModelInput { get }
+    var output: ShortenLinkViewModelOutput { get }
+}
+
+protocol ShortenLinkViewModelOutput {
+
+    var linkShortenSuccess: Driver<Bool> { get }
+}
+
+protocol ShortenLinkViewModelInput {
+
+    func shortenLink(link: String)
```

Comment:
You can bind the ` shortenButton.rx.tap` event directly if you use `PublishRelay<String>`

Reply:
This binding is the same way as our MVVM demo. The benefit is that the input is one way binding where output is another one way binding hence making the viewModel easier to read through.